fenix/app/src/main/java/org/mozilla/fenix/home/HomeFragment.kt

```diff
@@ -1007,17 +1014,23 @@ class HomeFragment : Fragment() {
                 lastAppliedWallpaperName = wallpaperName
             }
             else -> {
-                runBlockingIncrement {
+                viewLifecycleOwner.lifecycleScope.launch {
+                    val lifecycle = viewLifecycleOwner.lifecycle
                     // loadBitmap does file lookups based on name, so we don't need a fully
                     // qualified type to load the image
                     val wallpaper = Wallpaper.Default.copy(name = wallpaperName)
+                    logger.debug("loading bitmap")
                     val wallpaperImage =
                         requireComponents.useCases.wallpaperUseCases.loadBitmap(wallpaper)
                     wallpaperImage?.let {
+                        logger.debug("loaded bitmap")
                         it.scaleToBottomOfView(binding.wallpaperImageView)
                         binding.wallpaperImageView.isVisible = true
                         lastAppliedWallpaperName = wallpaperName
+                        logger.debug("applied bitmap")
                     } ?: run {
+                        logger.debug("bitmap null, is coroutine active $isActive, ${lifecycle.currentState}")
+                        if (!isActive) return@run
```

Comment:
As mode is updated on privacy mode to normal mode button click, the `observeWallpaperUpdates` gets triggered. Then the activity is recreated as part of themeManager logic, that results in this coroutine being cancelled as the view is destroyed, so we check here to make sure if the coroutine is active. If not, nothing to do.